### PR TITLE
Add production stage to Dockerfile

### PR DESCRIPTION
**Pull request by phuongdh**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,46 +1,48 @@
...
+
+# Download binaries in parallel
 FROM        ruby:2.5.5-stretch as download
+RUN         curl -L https://github.com/jwilder/dockerize/releases/download/v0.6.1/dockerize-linux-amd64-v0.6.1.tar.gz | tar xvz -C /usr/bin/
 RUN         curl https://chromedriver.storage.googleapis.com/2.46/chromedriver_linux64.zip -o /usr/local/bin/chromedriver \
-         && chmod +x /usr/local/bin/chromedriver \
-         && curl -L https://github.com/jwilder/dockerize/releases/download/v0.6.1/dockerize-linux-amd64-v0.6.1.tar.gz | tar xvz -C /usr/bin/ \
-         && mkdir -p /tmp/ffmpeg && cd /tmp/ffmpeg \
+         && chmod +x /usr/local/bin/chromedriver
+RUN         curl https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb -o /chrome.deb
```
Comment (cjcolvar) [inline Dockerfile]: Is this installed later or only downloaded?

Comment (cjcolvar) [reply Dockerfile]: Nvm...I see it now.

File: Dockerfile [hunk 2]
```diff
@@ -1,46 +1,48 @@
...
+RUN         mkdir -p /tmp/ffmpeg && cd /tmp/ffmpeg \
          && curl https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-amd64-static.tar.xz | tar xJ \
          && cp `find . -type f -executable` /usr/bin/
 
 
-# Dev stage for building dev image
-FROM        ruby:2.5.5-slim-stretch as dev
+# Base stage for building final images
+FROM        ruby:2.5.5-slim-stretch as base
 ENV         BUNDLER_VERSION 2.0.2
```
Comment (cjcolvar) [inline Dockerfile]: Should this be generalized like: https://github.com/avalonmediasystem/avalon/pull/4053/files#diff-3254677a7917c6c01f55212f86c57fbfR16

File: config/environments/production.rb [hunk 3]
```diff
@@ -60,16 +60,28 @@
...
+  config.log_formatter = ::Logger::Formatter.new
+
   # Enable logging to both stdout and file, in more compact format
-  config.lograge.enabled = true
-  config.lograge.custom_options = lambda do |event|
-    {:time => event.time}
+  if ENV["RAILS_LOG_TO_STDOUT"].present?
+    logger           = ActiveSupport::Logger.new(STDOUT)
+    logger.formatter = config.log_formatter
+    config.logger    = ActiveSupport::TaggedLogging.new(logger)
```
Comment (cjcolvar) [inline config/environments/production.rb]: 👍

**Comment by phuongdh** [comment]
> This looks good to me. I only had a couple questions.
> What impact does splitting some of the multi-line RUN commands into their own lines have? I thought you had collapsed them before to cut down on the number of layers.

Cutting down on layers only matters in the final image. For the supporting stages, I sometimes split them up to get granular caching (1 voided cache doesn't ruin others)

File: Dockerfile (hunk 2 above)
Comment (phuongdh) [reply Dockerfile]: True

**Pull request merged**